README.md

```diff
@@ -4,41 +4,162 @@ Please edit this README.md file with your answers to the worksheet questions.
+| 1 | 0 | 0     | 1     | 0               |
+| 1 | 1 | 0     | 0     | 0               |
 
 ### b
+| A | B | A AND B | NOT (A AND B) |
+|---|---|---------|---------------|
+| 0 | 0 | 0       | 1             |
+| 0 | 1 | 0       | 1             |
+| 1 | 0 | 0       | 1             |
+| 1 | 1 | 1       | 0             |
+
+| A | B | NOT A | NOT B | NOT A OR NOT B |
+|---|---|-------|-------|----------------|
+| 0 | 0 | 1     | 1     | 1              |
+| 0 | 1 | 1     | 0     | 1              |
+| 1 | 0 | 0     | 1     | 1              |
+| 1 | 1 | 0     | 0     | 0              |
 
 ### c
+| A | B | C | A AND B | A AND C | (A AND B) OR (A AND C) |
+|---|---|---|---------|---------|------------------------|
+| 0 | 0 | 0 | 0       | 0       | 0                      |
+| 0 | 1 | 0 | 0       | 0       | 0                      |
+| 1 | 0 | 0 | 0       | 0       | 0                      |
+| 1 | 1 | 0 | 1       | 0       | 1                      |
+| 0 | 0 | 1 | 0       | 0       | 0                      |
+| 0 | 1 | 1 | 0       | 1       | 1                      |
+| 1 | 0 | 1 | 0       | 1       | 1                      |
+| 1 | 1 | 1 | 1       | 1       | 1                      |
+
+| A | B | C | B OR C | A AND (B OR C) |
+|---|---|---|--------|----------------|
+| 0 | 0 | 0 | 0      | 0              |
+| 0 | 1 | 0 | 1      | 0              |
+| 1 | 0 | 0 | 0      | 0              |
+| 1 | 1 | 0 | 1      | 1              |
+| 0 | 0 | 1 | 1      | 0              |
+| 0 | 1 | 1 | 1      | 0              |
+| 1 | 0 | 1 | 1      | 1              |
+| 1 | 1 | 1 | 1      | 1              |
 
 ### d
+| A | B | C | A OR B | A OR C | (A OR B) AND (A OR C) |
+|---|---|---|--------|--------|-----------------------|
+| 0 | 0 | 0 | 0      | 0      | 0                     |
+| 0 | 1 | 0 | 1      | 0      | 0                     |
+| 1 | 0 | 0 | 1      | 1      | 1                     |
+| 1 | 1 | 0 | 1      | 1      | 1                     |
+| 0 | 0 | 1 | 0      | 1      | 0                     |
+| 0 | 1 | 1 | 1      | 1      | 1                     |
+| 1 | 0 | 1 | 1      | 1      | 1                     |
+| 1 | 1 | 1 | 1      | 1      | 1                     |
+
+| A | B | C | B AND C | A OR (B AND C) |
+|---|---|---|---------|----------------|
+| 0 | 0 | 0 | 0       | 0              |
+| 0 | 1 | 0 | 0       | 0              |
+| 1 | 0 | 0 | 0       | 1              |
+| 1 | 1 | 0 | 0       | 1              |
+| 0 | 0 | 1 | 0       | 0              |
+| 0 | 1 | 1 | 1       | 1              |
+| 1 | 0 | 1 | 0       | 1              |
+| 1 | 1 | 1 | 1       | 1              |
 
 ## Question 4
 
 ### a
+Same identity as Question 3 B
 
+The code: if not(file_exists("a.txt") and file_exists("b.txt"))
+Is the same as NOT (A AND B)
+
+The code: if not file_exists("a.txt") or not file_exists("b.txt")
+Is the same as NOT A OR NOT B
 ### b
 
+
 ### c
+Same identity as Question 3 B
+
+The code: if x == 0 and y == 0:
+Is the same as NOT (A AND B)
```

Comment:
Worth noting that the NOT here comes from the if and else being switched around between the two programs